webhooks/webhooks.go

```diff
@@ -253,10 +254,21 @@ func sendEvent(ctx context.Context, url string, action Event) error {
 		return err
 	}
 
-	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewReader(body))
+	parsedURL, err := nurl.Parse(url)
 	if err != nil {
 		return err
 	}
+
+	// extract username and password from the URL
```

Comment:
So the fact that this needs to be done actually confused me since it works fine in the browser.
After some digging it seems like using URLs with `username` and `password` in them was actually deprecated and only headers are now supported. Which might explain why the go http client doesn't just turn this into the `Authorization` header implicitly the way you do it here manually.

This makes me think it's probably better to respect that change and extend `Register` with `opts` that allow to specify basic authentication via `WithBasicAuth(username, password)` instead of implicitly parsing that from the URL.

To avoid the database migration we could squeeze it into the URL still for now or we add another json column for authentication right away. I'll leave that up to you.

Reply:
Hm ok so I added the `WithBasicAuth` thing. I hope it's more or less what you had in mind. I went back and forth quite a bit on how to add it. I ended up with the migration after all, I think if we don't use the user information we might as well bite the bullet and add the extra column. If we do add that column, I figured it's best to keep it a little generic and flexible, so it's a `JSON` column called headers. The `WithBasicAuth` is definitely useful now, in the future I think we could use the headers for debugging, or specifying the content type of the payload, maybe source info. Let me know what you think, I don't mind changing it again it's important to not add technical debt here I think.